Revised NavBar

Steve-Leke opened the pull request:
**Goal:** 
- We're trying to have one file that has all the navBar code and all the additional css/js link imports for the head, and then we inject it into every page. 

**Implementations:**
a) Use Jquery Method Referenced In The Issue
b) Create A NavBar HTML Element That's Injectable In All Pages
c) Turn Every Page Into An HTML Element and Call The Elements Inside The Index.html page
d) Then, Find A Way To Keep Track Of What Page Is Being Shown (Could Be Done Using active class)
e) Use a New Framework Such as Polymer, Angular, React, or Others

**Issues/Concerns:**
a) Which way is the easiest to implement? (In terms of learning curve/easy to edit)
b) Which implementation will last long / won't deprecate short-term?
c) Which implementation will work best with our upcoming Firebase/WebRTC features?
d) Which implementation has the fastest web performance? 
(Something to consider about (d) is cached data. If someone's been on our site before, they won't face major performance issues, compared to someone who's visiting the site for the first time.)

Steve-Leke committed:
Revised NavBar

j0n3lson commented:
Hi  @Steve-Leke, 

 A random law from [Hacker Laws repository](https://github.com/dwmkerr/hacker-laws) is a small gift to thank you for opening the PR.

-----------------------------------------------------
YAGNI
-----------------------------------------------------

YAGNI on Wikipedia

This is an acronym for _**Y**ou **A**in't **G**onna **N**eed **I**t_.

> Always implement things when you actually need them, never when you just foresee that you need them.
>
> ([Ron Jeffries](https://twitter.com/RonJeffries)) (XP co-founder and author of the book "Extreme Programming Installed")

This _Extreme Programming_ (XP) principle suggests developers should only implement functionality that is needed for the immediate requirements, and avoid attempts to predict the future by implementing functionality that might be needed later.

Adhering to this principle should reduce the amount of unused code in the codebase, and avoid time and effort being wasted on functionality that brings no value.

See also:

- Reading List: Extreme Programming Installed

-----------------------------------------------------
         github.com/dwmkerr/hacker-laws by Dave Kerr 
-----------------------------------------------------

Steve-Leke committed:
Implementation B

Steve-Leke commented:
In the recent commit (f25730b), I tried out Implementation B that's listed in the very first comment.

**Comments:**
- This implementation answers Issue/Concern A listed in the very fist comment.
- I implemented it only on the patientSignUp.html and the index.html page
- Implementing the navBar this way, fixed the navBar display issue at (#30).
- However, we lose having the sticky nav-bar when this method is being used.
(It may still be possible to have the sticky nav-bar, but I haven't figured out the best way to do so.)

lillianchou94 commented on public/templates/navBar.html:
```diff
@@ -0,0 +1,82 @@
+type="text/css"
+href="/assets/css/font-awesome.css"
+/>
+<link
+rel="stylesheet"
+type="text/css"
+href="/assets/css/templatemo-art-factory.css"
+/>
```
Can we double check that all these imports are needed for the header? From inspection, I only see templatemo-art-factory.css and bootstrap.min.css being applied

Steve-Leke replied:
Every page uses these imports differently. Some pages may use some and others won't. If we're going to generalize everything it may be easier to keep them all than create some kind of conditional checking for every page.

lillianchou94 commented on public/templates/index.html:
```diff
@@ -42,11 +42,13 @@
     <!-- jQuery -->
     <script defer src="/assets/js/jquery-2.1.0.min.js"></script>
 
+    <!-- jQuery -->
+    <scrip defer src="https://code.jquery.com/jquery-3.5.1.js"></script>
+
```
is ths intentional?

Steve-Leke replied:
Yeah, 3.5.1 is the newest version of Jquery but some pages are having issues with it and others aren't. I'm unsure if this is the way to go.

lillianchou94 commented on public/components/navBar.js:
```diff
@@ -0,0 +1,18 @@
+    }
+    customElements.define('nav-bar', NavBar);
+}
+
+
+
```
delete

lillianchou94 commented on public/components/navBar.js:
```diff
@@ -0,0 +1,18 @@
+    .then(stream => stream.text())
+    .then(text => label(text));
+
+function label(html) {
```
change the function name to something more self-documenting... defineElement? defineNavBar?

j0n3lson commented:
(pr-nagger-bot): Friendly ping. If you are a reviewer please take a look. If you are an author, please respond to comments.

lillianchou94 commented:
From offline discussion: Since the sticky/scrolling animation for the index page requires the header component to be on the main dom (and not in its own shadow dom), we can keep the header definition on index page as is. But reuse the custom element you defined in the additional pages that will link to the main page.

Steve-Leke commented:
Revert index.html to normal. Use the custom <navbar> tags on all the other pages though.

j0n3lson commented:
(pr-nagger-bot): Friendly ping. If you are a reviewer please take a look. If you are an author, please respond to comments.

Steve-Leke commented:
Wrong branch
